src/OpenLoco/src/Map/MapGenerator/OriginalTerrainGenerator.cpp

```diff
@@ -31,8 +31,8 @@ namespace OpenLoco::World::MapGenerator
         }
 
         // 0x00462718
-        const auto randX = (randomVal >> 14) & 0x1FF;
-        const auto randY = (randomVal >> 23) & 0x1FF;
+        const auto randX = (randomVal >> 14) % (heightMap.width - 1);
```

Comment:
Since this will have a different result compared to vanilla we might as well just change it to multiple calls to `getGameState().rng.randNext()` with the max set. Then we don't need to mess about with shifts.